### PR TITLE
Handle empty series data when calculating domains from data

### PR DESCRIPTION
**Pull request by jasmineleeson**
- also recalculate domains when the series data becomes non-empty

### CODE REVIEW

File: src/components/DataProvider/index.js [hunk 1]
```diff
@@ -13,6 +13,11 @@ export const calculateDomainFromData = (
   minAccessor = null,
   maxAccessor = null
 ) => {
+  // if there is no data, hard code the domain
+  if (!data || !data.length) {
+    return [-0.25, 0.25];
```
Comment (jasmineleeson) [inline src/components/DataProvider/index.js]: I don't really know what we should default to here

Comment (evancharlton) [reply src/components/DataProvider/index.js]: Seems as good as anything.

File: src/components/DataProvider/index.js [hunk 2]
```diff
@@ -447,7 +452,10 @@ export default class DataProvider extends Component {
       y1Accessor: seriesObject.y1Accessor,
     };
     const stateUpdates = {};
-    if (reason === 'MOUNTED') {
+    if (
+      reason === 'MOUNTED' ||
+      (!seriesObject.data.length && loaderConfig.data.length)
```
Comment (evancharlton) [inline src/components/DataProvider/index.js]:
```suggestion
      (seriesObject.data.length === 0 && loaderConfig.data.length > 0`)
```

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-220.surge.sh

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-220.surge.sh

**Pull request merged**